bindings/alicloud/rocketmq/rocketmq.go

```diff
@@ -0,0 +1,256 @@
+
+			return nil, fmt.Errorf("setupPublisher err:%w", err)
+		}
+
+		return producer, nil
+	}
+
+	return nil, errors.New("binding-rocketmq error: cannot found rocketmq producer")
+}
+
+// Operations returns list of operations supported by rocketmq binding
+func (t *AliCloudRocketMQ) Operations() []bindings.OperationKind {
+	return []bindings.OperationKind{bindings.CreateOperation}
+}
+
+func (t *AliCloudRocketMQ) Invoke(req *bindings.InvokeRequest) (*bindings.InvokeResponse, error) {
+	rst := &bindings.InvokeResponse{Data: nil, Metadata: nil}
+
+	if req.Operation != bindings.CreateOperation {
+		return rst, fmt.Errorf("binding-rocketmq error: unsupported operation %s", req.Operation)
+	}
+
+	return rst, t.sendMessage(req)
+}
+
+func (t *AliCloudRocketMQ) sendMessage(req *bindings.InvokeRequest) error {
+	topic := req.Metadata[metadataRocketmqTopic]
+
+	if topic != "" {
+		_, err := t.send(topic, req.Metadata[metadataRocketmqTag], req.Metadata[metadataRocketmqKey], req.Data)
+		if err != nil {
+			return err
+		}
+
+		return nil
+	}
+
+	for _, topicStr := range strings.Split(t.settings.Topics, topicSeparator) {
+		if topicStr == "" {
+			continue
+		}
+		_, mqExpression, topic, err := parseTopic(topicStr)
+		if err != nil {
+			return err
+		}
+		_, err = t.send(topic, mqExpression, req.Metadata[metadataRocketmqKey], req.Data)
+		if err != nil {
+			return err
+		}
+		t.logger.Debugf("binding-rocketmq send msg done, topic:%s tag:%s data-length:%d ", topic, mqExpression, len(req.Data))
+	}
+
+	return nil
+}
+
+func (t *AliCloudRocketMQ) send(topic, mqExpr, key string, data []byte) (bool, error) {
+	msg := primitive.NewMessage(topic, data).WithTag(mqExpr).WithKeys([]string{key})
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+	rst, err := t.producer.SendSync(ctx, msg)
+	if err != nil {
+		return false, fmt.Errorf("binding-rocketmq: send failed err:%w", err)
+	}
+	if rst.Status == 0 {
+		return true, nil
+	}
+
+	return false, fmt.Errorf("binding-rocketmq: unexpected status:%d", rst.Status)
+}
+
+type mqCallback func(ctx context.Context, msgs ...*primitive.MessageExt) (mqc.ConsumeResult, error)
+
+func (t *AliCloudRocketMQ) adaptCallback(_, consumerGroup, mqType, mqExpr string, handler func(*bindings.ReadResponse) ([]byte, error)) mqCallback {
+	return func(ctx context.Context, msgs ...*primitive.MessageExt) (mqc.ConsumeResult, error) {
+		var success = true
+		for _, v := range msgs {
+			metadata := make(map[string]string, 4)
+			metadata[metadataRocketmqType] = mqType
+			metadata[metadataRocketmqExpression] = mqExpr
+			metadata[metadataRocketmqConsumerGroup] = consumerGroup
+			if v.Queue != nil {
+				metadata[metadataRocketmqBrokerName] = v.Queue.BrokerName
+			}
+			t.logger.Debugf("binging-rocketmq handle msg, topic:%s msg-id:%s data-length:%d ", v.Topic, len(v.Body), v.MsgId)
+
+			msg := &bindings.ReadResponse{Data: v.Body,
+				Metadata: metadata}
+			if _, err := handler(msg); err != nil {
```

Comment:
I see the `mqc.ConsumeRetryLater` below but is there any case where you would want to retry using the `/internal/retry` package?